### PR TITLE
feat: return_allocated_node_data

### PR DESCRIPTION
**Pull request by kobayu858**
## Description

- Return_allocated_node_data
- For preemptive scheduler

## Related links

None.

## Pre-review checklist for the PR author

- [x] I've confirmed the [self-checklist](https://docs.google.com/spreadsheets/d/1p5KMcm752iCySHJz0xL1qf6fcLDKprbo/edit#gid=863618628).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: lib/src/core.rs [hunk 1]
```diff
@@ -58,6 +58,21 @@ impl Core {
         }
         Continue
     }
+
+    pub fn return_allocated_node_data(&mut self) -> Option<NodeData> {
```
Comment (atsushi421) [inline lib/src/core.rs:62]: How about `suspend_execution` ?

File: lib/src/homogeneous.rs [hunk 2]
```diff
@@ -33,6 +33,10 @@ impl ProcessorBase for HomogeneousProcessor {
         }
         None
     }
+
+    fn return_allocated_node_data(&mut self, core_id: usize) -> Option<NodeData> {
```
Comment (atsushi421) [inline lib/src/homogeneous.rs:37]: See above.

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/core.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/core.rs]: Changed.

File: lib/src/homogeneous.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/homogeneous.rs]: Changed.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Pull request merged**